dataframe-csv/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/DelimCsvTsvTests.kt

```diff
@@ -0,0 +1,657 @@
+import kotlinx.datetime.LocalDateTime
+import org.intellij.lang.annotations.Language
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.api.allNulls
+import org.jetbrains.kotlinx.dataframe.api.convert
+import org.jetbrains.kotlinx.dataframe.api.dataFrameOf
+import org.jetbrains.kotlinx.dataframe.api.group
+import org.jetbrains.kotlinx.dataframe.api.groupBy
+import org.jetbrains.kotlinx.dataframe.api.into
+import org.jetbrains.kotlinx.dataframe.api.isEmpty
+import org.jetbrains.kotlinx.dataframe.api.print
+import org.jetbrains.kotlinx.dataframe.api.schema
+import org.jetbrains.kotlinx.dataframe.api.toStr
+import org.junit.Test
+import java.io.File
+import java.io.StringWriter
+import java.math.BigDecimal
+import java.net.URL
+import java.util.Locale
+import java.util.zip.GZIPInputStream
+import kotlin.reflect.KClass
+import kotlin.reflect.typeOf
+
+@Suppress("ktlint:standard:argument-list-wrapping")
+class DelimCsvTsvTests {
+
+    @Test
+    fun readNulls() {
+        @Language("CSV")
+        val src =
+            """
+            first,second
+            2,,
+            3,,
+            """.trimIndent()
+        val df = DataFrame.readCsvStr(src)
+        df.rowsCount() shouldBe 2
+        df.columnsCount() shouldBe 2
+        df["first"].type() shouldBe typeOf<Int>()
+        df["second"].allNulls() shouldBe true
+        df["second"].type() shouldBe typeOf<String?>()
+    }
+
+    @Test
+    fun write() {
+        val df = dataFrameOf("col1", "col2")(
+            1, null,
+            2, null,
+        ).convert("col2").toStr()
+
+        val str = StringWriter()
+        df.writeCsv(str)
+
+        val res = DataFrame.readCsvStr(str.buffer.toString())
+
+        res shouldBe df
+    }
+
+    @Test
+    fun readCsv() {
+        val df = DataFrame.read(simpleCsv)
+
+        df.columnsCount() shouldBe 11
+        df.rowsCount() shouldBe 5
+        df.columnNames()[5] shouldBe "duplicate1"
+        df.columnNames()[6] shouldBe "duplicate11"
+        df["duplicate1"].type() shouldBe typeOf<Char?>()
+        df["double"].type() shouldBe typeOf<Double?>()
+        df["number"].type() shouldBe typeOf<Double>()
+        df["time"].type() shouldBe typeOf<LocalDateTime>()
+
+        df.print(columnTypes = true, borders = true, title = true)
+    }
+
+    @Test
+    fun `read ZIP Csv`() {
+        val df = DataFrame.readCsv(simpleCsvZip)
+
+        df.columnsCount() shouldBe 11
+        df.rowsCount() shouldBe 5
+        df.columnNames()[5] shouldBe "duplicate1"
+        df.columnNames()[6] shouldBe "duplicate11"
+        df["duplicate1"].type() shouldBe typeOf<Char?>()
+        df["double"].type() shouldBe typeOf<Double?>()
+        df["number"].type() shouldBe typeOf<Double>()
+        df["time"].type() shouldBe typeOf<LocalDateTime>()
+
+        df.print(columnTypes = true, borders = true, title = true)
+    }
+
+    @Test
+    fun `read GZ Csv`() {
+        val df = DataFrame.readCsv(simpleCsvGz)
```

Comment:
I think a lot of these assertions aren't needed. We only need to test that reading .gz works the same as reading plain csv

Reply:
yeah that's fair